libdevice/device_imf.hpp

```diff
@@ -16,6 +16,11 @@
 
 #ifdef __LIBDEVICE_IMF_ENABLED__
 
+#if !defined(__SPIR__) && !defined(__LIBDEVICE_HOST_IMPL__)
```

Comment:
Hi, @steffenlarsen 
Does this change meet your expectation?
Thanks very much.

Reply:
Yes, looks good. Thank you!